reanalysis/query_panelapp.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python3
+
+
+"""
+PanelApp Parser for Reanalysis
+
+Takes a panel ID
+For the latest content, pulls Symbol, ENSG, and MOI
+    (MOI is simplified from PanelApp enum)
+
+Optionally user can provide a date in the past
+Identify the highest panel version prior to that date
+Pull all details from the earlier version
+Store all discrepancies between earlier and current
+
+Write all output to a JSON dictionary
+"""
+
+
+import logging
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union, Set
+import json
+
+from google.cloud import storage
+import requests
+
+import click
+
+
+# panelapp URL constants
+PANELAPP_ROOT = 'https://panelapp.agha.umccr.org/api/v1'
+PANEL_ROOT = f'{PANELAPP_ROOT}/panels'
+PANEL_CONTENT = f'{PANEL_ROOT}/{{panel_id}}'
+ACTIVITIES = f'{PANEL_CONTENT}/activities'
+
+
+def get_gcp_blob(bucket_path: str) -> storage.blob:
```

Comment:
This would be a good candidate for using https://github.com/drivendataorg/cloudpathlib.